blockchain_connector/common/connector_common/base_connector.py

```diff
@@ -81,24 +84,32 @@ def sync_registries(self):
         # TODO need to use registries.
         logging.info("Not implemented yet")
 
-    def sync_workers(self):
+    async def sync_workers(self):
         """
         Sync up workers between blockchain and avalon
         """
         # Fetch first worker details from shared KV (via direct API)
         # and add the worker to block chain.
-        # TODO: Fetch all workers from shared KV and block chain
-        # and do 2-way sync.
-        self._active_worker_ids = \
-            self._worker_delegate.lookup_workers_in_kv_storage()
-        # Lookup worker ids from blockchain
-        chain_worker_ids = self._worker_delegate.lookup_workers_onchain()
-        # Add/Update workers to chain
-        self._worker_delegate.add_update_worker_to_chain(
-            self._active_worker_ids,
-            chain_worker_ids)
-
-    def sync_work_orders(self):
+        # TODO: Fetch all workers from blockchain and update kv store
+        # That will be a 2-way sync. As of now, only one way.
+        while True:
+            try:
+                self._active_worker_ids = \
+                    self._worker_delegate.lookup_workers_in_kv_storage()
+                # Lookup worker ids from blockchain
+                chain_worker_ids = self._worker_delegate.\
+                    lookup_workers_onchain()
+                # Add/Update workers to chain
+                self._worker_delegate.add_update_worker_to_chain(
```

Comment:
This loop will to call worker_update every time for the same worker. https://github.com/hyperledger/avalon/blob/f4d9f6a3b984f3c1c4114afb73d4c29068af1ef8/blockchain_connector/common/connector_common/worker_delegate.py#L104

Reply:
Yes @Ram-srini we had a discussion over this. We need to build the intelligence to not update if things haven't changed. A simple way would be to hold a map of workers in-memory and check against it. We could calculate a hash of the JSON string (representing a worker from KV) to avoid field by field matching. 
Then at every poll, we can simply fetch worker from kv & match hashes.
```
dict = {} # (worker_id->hash)
for( w_id in kv):
    w_details_json = kv.retrieve(wo_id)
    w_hash = _get_hash(json.dumps(w_details_json))
    if w_id not in dict:
        w_registry.add(w_id, w_details_json)
    else:
        if w_hash != dict[w_id]:
            w_registry.update(w_id, w_details_json)
            w_registry.set_status(w_id, w_details_json["status"])
```

Reply:
Do you want to implement it in a separate PR?

Reply:
No @Ram-srini . Will update this PR itself.